src/Foundation/NSUrlSessionHandler.cs

```diff
@@ -1083,6 +1089,18 @@ void DidReceiveChallengeImpl (NSUrlSession session, NSUrlSessionTask task, NSUrl
 					}
 					return;
 				}
+                if (sessionHandler.ClientCertificateOptions == ClientCertificateOption.Manual && challenge.ProtectionSpace.AuthenticationMethod == NSUrlProtectionSpace.AuthenticationMethodClientCertificate)
+                {
+					var certificate = System.Net.Security.CertificateHelper.GetEligibleClientCertificate(sessionHandler.ClientCertificates);
```

Comment:
Hm, is there something else tied to this property that works with the collection as a whole? Do you need a collection of client certificates if only the first certificate that passes some KU and EKU checks is used?

Reply:
This is the logic httpclienthandler and socketshttphandler uses. I’ve honestly never seen a reason to add multiple certificates here but logic is first eligible certificate wins